qstack/spahm/rho/bond.py

```diff
@@ -90,29 +87,38 @@ def main():
         xyzlist = utils.get_xyzlist(xyzlistfile)
         charge  = utils.get_chsp(args.charge, len(xyzlist))
         spin    = utils.get_chsp(args.spin,   len(xyzlist))
-    mols    = utils.load_mols(xyzlist, charge, spin, args.basis, args.print, units=args.units)
-
-    allvec = get_repr(mols, xyzlist, args.guess, xc=args.xc, spin=args.spin, readdm=args.readdm, printlevel=args.print,
-                      pairfile=args.pairfile, dump_and_exit=args.dump_and_exit,
-                      bpath=args.bpath, cutoff=args.cutoff, omods=args.omod,
-                      elements=args.elements, only_m0=args.only_m0, zeros=args.zeros, split=args.split)
-
-
-
-    if len(allvec) == 1:
-        allvec = allvec[0]
+    mols    = utils.load_mols(xyzlist, charge, spin, args.basis, args.print, units=args.units, ecp=args.ecp)
+    if args.with_symbols:
+        if len(args.only_z) > 0:
+            all_atoms   = np.array([z for mol in mols for z in mol.elements if z in args.only_z]).flatten()
+        else:
+            all_atoms   = np.array([mol.elements for mol in mols]).flatten()
+    if not args.dump_and_exit:
+        dms     = utils.mols_guess(mols, xyzlist, args.guess,
+                               xc=defaults.xc, spin=args.spin, readdm=args.readdm, printlevel=args.print)
+    else:
+        dms = []
+    allvec  = bond(mols, dms, args.bpath, args.cutoff, args.omod,
+                   spin=args.spin, elements=args.elements,
+                   only_m0=args.only_m0, zeros=args.zeros, split=args.split, printlevel=args.print,
+                   pairfile=args.pairfile, dump_and_exit=args.dump_and_exit, same_basis=args.same_basis, only_z=args.only_z)
 
     if args.print>1: print(allvec.shape)
 
+    allvec = np.squeeze(allvec)
+    allvec = np.array(allvec, ndmin=2)
     if args.spin:
         if args.merge is False:
             for omod, vec in zip(args.omod, allvec):
+                if args.with_symbols: allvec = np.array([(z, v) for v,z in zip(allvec, all_atoms)], dtype=object)
                 np.save(args.name_out+'_'+omod, vec)
         else:
-            np.save(args.name_out+'_'+'_'.join(args.omod), np.hstack(allvec))
+            allvec = np.hstack(allvec)
+            if args.with_symbols: allvec = np.array([(z, v) for v,z in zip(allvec, all_atoms)], dtype=object)
+            np.save(args.name_out+'_'+'_'.join(args.omod), allvec)
     else:
-        np.save(args.name_out, allvec)
-
+        if args.with_symbols: allvec = np.array([(z, v) for v,z in zip(allvec, all_atoms)], dtype=object)
+        np.save(args.name_out+'_'+'_'.join(args.omod), allvec)
```

Comment:
This part is a bit convoluted!

The default (and only) behaviour in spahm-a is to always return tuple `(symbols, reps, omod)` and then the `main()` deals with the merging. _and there's no option to save without symbols_

- maybe it's worth following spahm-a idea where the specific function for each local rep returns the tuple (with everything separated) and let the `main()` function deal with the symbols and the merging options.  
_What do you think?_

Reply:
I don't think it should be main. Can be get_repr that you removed.
The function can be run from a python script so almost everything meaningful should be in functions